contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,239 @@
+	handler := creat(r.root, path, mode)
+	if handler == nil {
+		return nil, handleErr(pointerOnRoot, r.root)
+	}
+
+	return &Handle{handle: handler}, nil
+}
+
+// Rename renames src to dst, or change flags on this file
+// if the names are the same it's only change the flags
+func (r *Root) Rename(src, dst string, flags int) error {
+	status := rename(r.root, src, dst, flags)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Mkdir creates a directory with a such mode by path
+func (r *Root) Mkdir(path string, mode uint) error {
+	status := mkdir(r.root, path, mode)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Mknod creates a filesystem node named path
+// with attributes mode and dev
+func (r *Root) Mknod(path string, mode uint, dev int) error {
+	status := mknod(r.root, path, mode, dev)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Hardlink creates a hardlink of file named target and place it to path
+func (r *Root) Hardlink(path, target string) error {
+	status := hardlink(r.root, path, target)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Symlink creates a symlink of file named target and place it to path
+func (r *Root) Symlink(path, target string) error {
+	status := symlink(r.root, path, target)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Free frees underling caught resources
+func (r *Root) Free() {
+	/*
+		Should we handle it in this way or always return not nil pointer on root?
+	*/
+	if r != nil {
+		free(pointerOnRoot, r.root)
+	}
+}
+
+// Handle represents an handle api of rspath
+type Handle struct {
+	handle cRootPtr
+}
+
+// Free frees underling caught resources
+func (h *Handle) Free() {
+	if h == nil {
+		return
+	}
+
+	free(pointerOnHandle, h.handle)
+}
+
+// Reopen upgrade the handle to a file representation
+func (h *Handle) Reopen(flags int) (*os.File, error) {
+	fd := reopen(h.handle, flags)
+	// TODO: verify what is correct path to handle this type of errors
+	// must it always be called error handling as in cat.c file or we can check fd first
+	if err := handleErr(pointerOnHandle, h.handle); err != nil {
+		return nil, err
+	}
+
+	file := os.NewFile(uintptr(fd), "")
+	return file, nil
+}
+
+// Error representation of rust error
+// particularly useful to not frighten to lost controll of pointer which can be rewritten.
+type Error struct {
+	description string
+	backtrace   []backtraceLine
+}
+
+type backtraceLine struct {
+	ip       uintptr
+	sAddress uintptr
+	sName    string
+	sFile    string
+	sLineno  uint32
+}
+
+func (err *Error) Error() string {
+	return err.description
+}
+
+// Backtrace flush backtrace of underlying error to string.
+//
+// Its not passed to realization of Error interface on purpose since
+// the main error should remain clear and simple
+func (err *Error) Backtrace() string {
```

Comment:
I will need to think some more on whether this backtrace interface is ideal -- I'd really want to be as compatible as possible with `github.com/pkg/errors`.

Reply:
Oh I forgot we don't have this https://github.com/pkg/errors/issues/216...